include/swift/Sema/ConstraintSystem.h

```diff
@@ -348,23 +348,45 @@ class TypeVariableType::Implementation {
     assert(getTypeVariable()->Bits.TypeVariableType.Options == bits
            && "Truncation");
   }
-
-  /// Whether this type variable can bind to an lvalue type.
-  bool canBindToLValue() const { return getRawOptions() & TVO_CanBindToLValue; }
-
-  /// Whether this type variable can bind to an inout type.
-  bool canBindToInOut() const { return getRawOptions() & TVO_CanBindToInOut; }
-
-  /// Whether this type variable can bind to an inout type.
-  bool canBindToNoEscape() const { return getRawOptions() & TVO_CanBindToNoEscape; }
-
-  /// Whether this type variable can bind to a hole.
-  bool canBindToHole() const { return getRawOptions() & TVO_CanBindToHole; }
-
-  /// Whether this type variable prefers a subtype binding over a supertype
-  /// binding.
-  bool prefersSubtypeBinding() const {
-    return getRawOptions() & TVO_PrefersSubtypeBinding;
+  
+  StringRef getTypeVariableOptions(TypeVariableOptions TVO) const {
+  #define ENTRY(Kind, String) case TypeVariableOptions::Kind: return String
+    switch (TVO) {
+    ENTRY(TVO_CanBindToLValue, "lvalue");
+    ENTRY(TVO_CanBindToInOut, "inout");
+    ENTRY(TVO_CanBindToNoEscape, "noescape");
+    ENTRY(TVO_CanBindToHole, "hole");
+    ENTRY(TVO_PrefersSubtypeBinding, "");
+    }
+  #undef ENTRY
+  }
+
+  /// Collect possible \c TypeVariableOptions bindings for this type variable
+  ///
+  /// \c skipSubtypeCheck ignores this check while printing type variables
+  /// binding options
+  SmallVector<TypeVariableOptions, 5>
+  getBindingOptions(bool skipSubtypeCheck = false) const {
```

Comment:
I thought we agreed that this might not be the best idea because it doesn't really buy as much in terms of safety - any new attribute would have to be added as a check here is easy to miss and can have pretty bad implications...